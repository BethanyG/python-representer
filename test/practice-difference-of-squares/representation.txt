Module([FunctionDef('placeholder_0', arguments([], [arg('placeholder_1', lineno=1, col_offset=18, end_lineno=1, end_col_offset=24)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_2', Store(), lineno=2, col_offset=4, end_lineno=2, end_col_offset=8)], BinOp(BinOp(Name('placeholder_1', Load(), lineno=2, col_offset=11, end_lineno=2, end_col_offset=17), Mult(), BinOp(Name('placeholder_1', Load(), lineno=2, col_offset=21, end_lineno=2, end_col_offset=27), Add(), Constant(1, lineno=2, col_offset=30, end_lineno=2, end_col_offset=31), lineno=2, col_offset=21, end_lineno=2, end_col_offset=31), lineno=2, col_offset=11, end_lineno=2, end_col_offset=32), Div(), Constant(2, lineno=2, col_offset=35, end_lineno=2, end_col_offset=36), lineno=2, col_offset=11, end_lineno=2, end_col_offset=36), lineno=2, col_offset=4, end_lineno=2, end_col_offset=36), Return(BinOp(Name('placeholder_2', Load(), lineno=3, col_offset=11, end_lineno=3, end_col_offset=15), Mult(), Name('placeholder_2', Load(), lineno=3, col_offset=18, end_lineno=3, end_col_offset=22), lineno=3, col_offset=11, end_lineno=3, end_col_offset=22), lineno=3, col_offset=4, end_lineno=3, end_col_offset=22)], [], lineno=1, col_offset=0, end_lineno=3, end_col_offset=22), FunctionDef('placeholder_3', arguments([], [arg('placeholder_1', lineno=6, col_offset=19, end_lineno=6, end_col_offset=25)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Name('placeholder_4', Store(), lineno=7, col_offset=4, end_lineno=7, end_col_offset=13)], BinOp(BinOp(Name('placeholder_1', Load(), lineno=7, col_offset=16, end_lineno=7, end_col_offset=22), Mult(), BinOp(Name('placeholder_1', Load(), lineno=7, col_offset=26, end_lineno=7, end_col_offset=32), Add(), Constant(1, lineno=7, col_offset=35, end_lineno=7, end_col_offset=36), lineno=7, col_offset=26, end_lineno=7, end_col_offset=36), lineno=7, col_offset=16, end_lineno=7, end_col_offset=37), Mult(), BinOp(BinOp(Constant(2, lineno=7, col_offset=41, end_lineno=7, end_col_offset=42), Mult(), Name('placeholder_1', Load(), lineno=7, col_offset=45, end_lineno=7, end_col_offset=51), lineno=7, col_offset=41, end_lineno=7, end_col_offset=51), Add(), Constant(1, lineno=7, col_offset=54, end_lineno=7, end_col_offset=55), lineno=7, col_offset=41, end_lineno=7, end_col_offset=55), lineno=7, col_offset=16, end_lineno=7, end_col_offset=56), lineno=7, col_offset=4, end_lineno=7, end_col_offset=56), Return(BinOp(Name('placeholder_4', Load(), lineno=8, col_offset=11, end_lineno=8, end_col_offset=20), Div(), Constant(6, lineno=8, col_offset=23, end_lineno=8, end_col_offset=24), lineno=8, col_offset=11, end_lineno=8, end_col_offset=24), lineno=8, col_offset=4, end_lineno=8, end_col_offset=24)], [], lineno=6, col_offset=0, end_lineno=8, end_col_offset=24), FunctionDef('placeholder_5', arguments([], [arg('placeholder_1', lineno=11, col_offset=26, end_lineno=11, end_col_offset=32)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(BinOp(Call(Name('placeholder_0', Load(), lineno=12, col_offset=11, end_lineno=12, end_col_offset=24), [Name('placeholder_1', Load(), lineno=12, col_offset=25, end_lineno=12, end_col_offset=31)], [], lineno=12, col_offset=11, end_lineno=12, end_col_offset=32), Sub(), Call(Name('placeholder_3', Load(), lineno=12, col_offset=35, end_lineno=12, end_col_offset=49), [Name('placeholder_1', Load(), lineno=12, col_offset=50, end_lineno=12, end_col_offset=56)], [], lineno=12, col_offset=35, end_lineno=12, end_col_offset=57), lineno=12, col_offset=11, end_lineno=12, end_col_offset=57), lineno=12, col_offset=4, end_lineno=12, end_col_offset=57)], [], lineno=11, col_offset=0, end_lineno=12, end_col_offset=57)], [])