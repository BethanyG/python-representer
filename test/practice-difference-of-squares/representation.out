# BEGIN TREE BEFORE
Module(
  [
    FunctionDef(
      'square_of_sum',
      arguments(
        [],
        [
          arg(
            'number',
            lineno=1,
            col_offset=18,
            end_lineno=1,
            end_col_offset=24)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'sum_',
              Store(),
              lineno=2,
              col_offset=4,
              end_lineno=2,
              end_col_offset=8)],
          BinOp(
            BinOp(
              Name(
                'number',
                Load(),
                lineno=2,
                col_offset=11,
                end_lineno=2,
                end_col_offset=17),
              Mult(),
              BinOp(
                Name(
                  'number',
                  Load(),
                  lineno=2,
                  col_offset=21,
                  end_lineno=2,
                  end_col_offset=27),
                Add(),
                Constant(
                  1,
                  lineno=2,
                  col_offset=30,
                  end_lineno=2,
                  end_col_offset=31),
                lineno=2,
                col_offset=21,
                end_lineno=2,
                end_col_offset=31),
              lineno=2,
              col_offset=11,
              end_lineno=2,
              end_col_offset=32),
            Div(),
            Constant(
              2,
              lineno=2,
              col_offset=35,
              end_lineno=2,
              end_col_offset=36),
            lineno=2,
            col_offset=11,
            end_lineno=2,
            end_col_offset=36),
          lineno=2,
          col_offset=4,
          end_lineno=2,
          end_col_offset=36),
        Return(
          BinOp(
            Name(
              'sum_',
              Load(),
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=15),
            Mult(),
            Name(
              'sum_',
              Load(),
              lineno=3,
              col_offset=18,
              end_lineno=3,
              end_col_offset=22),
            lineno=3,
            col_offset=11,
            end_lineno=3,
            end_col_offset=22),
          lineno=3,
          col_offset=4,
          end_lineno=3,
          end_col_offset=22)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=3,
      end_col_offset=22),
    FunctionDef(
      'sum_of_squares',
      arguments(
        [],
        [
          arg(
            'number',
            lineno=6,
            col_offset=19,
            end_lineno=6,
            end_col_offset=25)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'numerator',
              Store(),
              lineno=7,
              col_offset=4,
              end_lineno=7,
              end_col_offset=13)],
          BinOp(
            BinOp(
              Name(
                'number',
                Load(),
                lineno=7,
                col_offset=16,
                end_lineno=7,
                end_col_offset=22),
              Mult(),
              BinOp(
                Name(
                  'number',
                  Load(),
                  lineno=7,
                  col_offset=26,
                  end_lineno=7,
                  end_col_offset=32),
                Add(),
                Constant(
                  1,
                  lineno=7,
                  col_offset=35,
                  end_lineno=7,
                  end_col_offset=36),
                lineno=7,
                col_offset=26,
                end_lineno=7,
                end_col_offset=36),
              lineno=7,
              col_offset=16,
              end_lineno=7,
              end_col_offset=37),
            Mult(),
            BinOp(
              BinOp(
                Constant(
                  2,
                  lineno=7,
                  col_offset=41,
                  end_lineno=7,
                  end_col_offset=42),
                Mult(),
                Name(
                  'number',
                  Load(),
                  lineno=7,
                  col_offset=45,
                  end_lineno=7,
                  end_col_offset=51),
                lineno=7,
                col_offset=41,
                end_lineno=7,
                end_col_offset=51),
              Add(),
              Constant(
                1,
                lineno=7,
                col_offset=54,
                end_lineno=7,
                end_col_offset=55),
              lineno=7,
              col_offset=41,
              end_lineno=7,
              end_col_offset=55),
            lineno=7,
            col_offset=16,
            end_lineno=7,
            end_col_offset=56),
          lineno=7,
          col_offset=4,
          end_lineno=7,
          end_col_offset=56),
        Return(
          BinOp(
            Name(
              'numerator',
              Load(),
              lineno=8,
              col_offset=11,
              end_lineno=8,
              end_col_offset=20),
            Div(),
            Constant(
              6,
              lineno=8,
              col_offset=23,
              end_lineno=8,
              end_col_offset=24),
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=24),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=24)],
      [],
      lineno=6,
      col_offset=0,
      end_lineno=8,
      end_col_offset=24),
    FunctionDef(
      'difference_of_squares',
      arguments(
        [],
        [
          arg(
            'number',
            lineno=11,
            col_offset=26,
            end_lineno=11,
            end_col_offset=32)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          BinOp(
            Call(
              Name(
                'square_of_sum',
                Load(),
                lineno=12,
                col_offset=11,
                end_lineno=12,
                end_col_offset=24),
              [
                Name(
                  'number',
                  Load(),
                  lineno=12,
                  col_offset=25,
                  end_lineno=12,
                  end_col_offset=31)],
              [],
              lineno=12,
              col_offset=11,
              end_lineno=12,
              end_col_offset=32),
            Sub(),
            Call(
              Name(
                'sum_of_squares',
                Load(),
                lineno=12,
                col_offset=35,
                end_lineno=12,
                end_col_offset=49),
              [
                Name(
                  'number',
                  Load(),
                  lineno=12,
                  col_offset=50,
                  end_lineno=12,
                  end_col_offset=56)],
              [],
              lineno=12,
              col_offset=35,
              end_lineno=12,
              end_col_offset=57),
            lineno=12,
            col_offset=11,
            end_lineno=12,
            end_col_offset=57),
          lineno=12,
          col_offset=4,
          end_lineno=12,
          end_col_offset=57)],
      [],
      lineno=11,
      col_offset=0,
      end_lineno=12,
      end_col_offset=57)],
  [])

# BEGIN TREE AFTER
Module(
  [
    FunctionDef(
      'placeholder_0',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=1,
            col_offset=18,
            end_lineno=1,
            end_col_offset=24)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_2',
              Store(),
              lineno=2,
              col_offset=4,
              end_lineno=2,
              end_col_offset=8)],
          BinOp(
            BinOp(
              Name(
                'placeholder_1',
                Load(),
                lineno=2,
                col_offset=11,
                end_lineno=2,
                end_col_offset=17),
              Mult(),
              BinOp(
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=2,
                  col_offset=21,
                  end_lineno=2,
                  end_col_offset=27),
                Add(),
                Constant(
                  1,
                  lineno=2,
                  col_offset=30,
                  end_lineno=2,
                  end_col_offset=31),
                lineno=2,
                col_offset=21,
                end_lineno=2,
                end_col_offset=31),
              lineno=2,
              col_offset=11,
              end_lineno=2,
              end_col_offset=32),
            Div(),
            Constant(
              2,
              lineno=2,
              col_offset=35,
              end_lineno=2,
              end_col_offset=36),
            lineno=2,
            col_offset=11,
            end_lineno=2,
            end_col_offset=36),
          lineno=2,
          col_offset=4,
          end_lineno=2,
          end_col_offset=36),
        Return(
          BinOp(
            Name(
              'placeholder_2',
              Load(),
              lineno=3,
              col_offset=11,
              end_lineno=3,
              end_col_offset=15),
            Mult(),
            Name(
              'placeholder_2',
              Load(),
              lineno=3,
              col_offset=18,
              end_lineno=3,
              end_col_offset=22),
            lineno=3,
            col_offset=11,
            end_lineno=3,
            end_col_offset=22),
          lineno=3,
          col_offset=4,
          end_lineno=3,
          end_col_offset=22)],
      [],
      lineno=1,
      col_offset=0,
      end_lineno=3,
      end_col_offset=22),
    FunctionDef(
      'placeholder_3',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=6,
            col_offset=19,
            end_lineno=6,
            end_col_offset=25)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Assign(
          [
            Name(
              'placeholder_4',
              Store(),
              lineno=7,
              col_offset=4,
              end_lineno=7,
              end_col_offset=13)],
          BinOp(
            BinOp(
              Name(
                'placeholder_1',
                Load(),
                lineno=7,
                col_offset=16,
                end_lineno=7,
                end_col_offset=22),
              Mult(),
              BinOp(
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=7,
                  col_offset=26,
                  end_lineno=7,
                  end_col_offset=32),
                Add(),
                Constant(
                  1,
                  lineno=7,
                  col_offset=35,
                  end_lineno=7,
                  end_col_offset=36),
                lineno=7,
                col_offset=26,
                end_lineno=7,
                end_col_offset=36),
              lineno=7,
              col_offset=16,
              end_lineno=7,
              end_col_offset=37),
            Mult(),
            BinOp(
              BinOp(
                Constant(
                  2,
                  lineno=7,
                  col_offset=41,
                  end_lineno=7,
                  end_col_offset=42),
                Mult(),
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=7,
                  col_offset=45,
                  end_lineno=7,
                  end_col_offset=51),
                lineno=7,
                col_offset=41,
                end_lineno=7,
                end_col_offset=51),
              Add(),
              Constant(
                1,
                lineno=7,
                col_offset=54,
                end_lineno=7,
                end_col_offset=55),
              lineno=7,
              col_offset=41,
              end_lineno=7,
              end_col_offset=55),
            lineno=7,
            col_offset=16,
            end_lineno=7,
            end_col_offset=56),
          lineno=7,
          col_offset=4,
          end_lineno=7,
          end_col_offset=56),
        Return(
          BinOp(
            Name(
              'placeholder_4',
              Load(),
              lineno=8,
              col_offset=11,
              end_lineno=8,
              end_col_offset=20),
            Div(),
            Constant(
              6,
              lineno=8,
              col_offset=23,
              end_lineno=8,
              end_col_offset=24),
            lineno=8,
            col_offset=11,
            end_lineno=8,
            end_col_offset=24),
          lineno=8,
          col_offset=4,
          end_lineno=8,
          end_col_offset=24)],
      [],
      lineno=6,
      col_offset=0,
      end_lineno=8,
      end_col_offset=24),
    FunctionDef(
      'placeholder_5',
      arguments(
        [],
        [
          arg(
            'placeholder_1',
            lineno=11,
            col_offset=26,
            end_lineno=11,
            end_col_offset=32)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          BinOp(
            Call(
              Name(
                'placeholder_0',
                Load(),
                lineno=12,
                col_offset=11,
                end_lineno=12,
                end_col_offset=24),
              [
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=12,
                  col_offset=25,
                  end_lineno=12,
                  end_col_offset=31)],
              [],
              lineno=12,
              col_offset=11,
              end_lineno=12,
              end_col_offset=32),
            Sub(),
            Call(
              Name(
                'placeholder_3',
                Load(),
                lineno=12,
                col_offset=35,
                end_lineno=12,
                end_col_offset=49),
              [
                Name(
                  'placeholder_1',
                  Load(),
                  lineno=12,
                  col_offset=50,
                  end_lineno=12,
                  end_col_offset=56)],
              [],
              lineno=12,
              col_offset=35,
              end_lineno=12,
              end_col_offset=57),
            lineno=12,
            col_offset=11,
            end_lineno=12,
            end_col_offset=57),
          lineno=12,
          col_offset=4,
          end_lineno=12,
          end_col_offset=57)],
      [],
      lineno=11,
      col_offset=0,
      end_lineno=12,
      end_col_offset=57)],
  [])

# BEGIN NORMALIZED CODE AFTER
def placeholder_0(placeholder_1):
    placeholder_2 = placeholder_1 * (placeholder_1 + 1) / 2
    return placeholder_2 * placeholder_2


def placeholder_3(placeholder_1):
    placeholder_4 = placeholder_1 * (placeholder_1 + 1) * (2 * placeholder_1 + 1)
    return placeholder_4 / 6


def placeholder_5(placeholder_1):
    return placeholder_0(placeholder_1) - placeholder_3(placeholder_1)
